src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -396,67 +403,112 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
+			addCodeCellButton.cellType = CellTypes.Code;
+
+			let addTextCellButton = new AddCellAction('notebook.AddTextCell', localize('text', "Markdown cell"), 'notebook-button masked-pseudo markdown');
+			addTextCellButton.cellType = CellTypes.Markdown;
+
+
+			this._runAllCellsAction = this.instantiationService.createInstance(RunAllCellsAction, 'notebook.runAllCells', localize('runAll', "Run all"), 'notebook-button masked-pseudo start-outline');
+
+			let collapseCellsAction = this.instantiationService.createInstance(CollapseCellsAction, 'notebook.collapseCells', true);
+
+			let clearResultsButton = new ClearAllOutputsAction('notebook.ClearAllOutputs', true);
+
+			this._trustedAction = this.instantiationService.createInstance(TrustedAction, 'notebook.Trusted', true);
+			this._trustedAction.enabled = false;
+
+			let taskbar = <HTMLElement>this.toolbar.nativeElement;
+			this._actionBar = new Taskbar(taskbar, { actionViewItemProvider: action => this.actionItemProvider(action as Action) });
+			this._actionBar.context = this;
+			taskbar.classList.add('in-preview');
+
+			let buttonDropdownContainer = DOM.$('li.action-item');
+			buttonDropdownContainer.setAttribute('role', 'presentation');
+			let dropdownMenuActionViewItem = new DropdownMenuActionViewItem(
+				addCodeCellButton,
+				[addCodeCellButton, addTextCellButton],
+				this.contextMenuService,
+				undefined,
+				this._actionBar.actionRunner,
+				undefined,
+				'codicon notebook-button masked-pseudo masked-pseudo-after add-new dropdown-arrow',
+				localize('addCell', "Cell"),
+				undefined
+			);
+			dropdownMenuActionViewItem.render(buttonDropdownContainer);
+			dropdownMenuActionViewItem.setActionContext(this);
+
+			this._actionBar.setContent([
+				{ element: buttonDropdownContainer },
+				{ action: this._runAllCellsAction },
+				{ element: Taskbar.createTaskbarSeparator() },
+				{ element: attachToContainer },
+				{ element: kernelContainer },
+				{ element: spacerElement },
+				{ action: collapseCellsAction },
+				{ action: clearResultsButton },
+				{ action: this._trustedAction },
+			]);
+		} else {
+			let kernelContainer = document.createElement('div');
+			let kernelDropdown = this.instantiationService.createInstance(KernelsDropdown, kernelContainer, this.contextViewService, this.modelReady);
+			kernelDropdown.render(kernelContainer);
+			attachSelectBoxStyler(kernelDropdown, this.themeService);
+
+			let attachToContainer = document.createElement('div');
+			let attachToDropdown = new AttachToDropdown(attachToContainer, this.contextViewService, this.modelReady,
+				this.connectionManagementService, this.connectionDialogService, this.notificationService, this.capabilitiesService);
+			attachToDropdown.render(attachToContainer);
+			attachSelectBoxStyler(attachToDropdown, this.themeService);
+
+			let addCodeCellButton = new AddCellAction('notebook.AddCodeCell', localize('code1', "Code"), 'notebook-button icon-add');
```

Comment:
So it's probably not an issue here because these strings are new - but in the future be really careful about updating strings like this. You could get into a situation where the localized files still have the old string for this ID and thus display the wrong thing for the new feature.

So in general you should never modify an ID after creating it - instead if you're adding new ones just make a new unique ID. I'd suggest you do that here.

Reply:
I get it. Thanks for bringing this up!

Reply:
Fixed